### PR TITLE
Updated spot-grid.py script

### PR DESCRIPTION
**Pull request by effting**
- Supports to load image from command line
- Indicator of first spot in the image
- 'Snake' to indicate order of spots

### CODE REVIEW

File: scripts/spot-grid.py [hunk 1]
```diff
@@ -168,6 +191,27 @@ def image_update(imp, window):
         logging.debug("Display thread ended")
 
 
+class StaticImageComponent(model.DigitalCamera):
+
+    def __init__(self, filename):
```
Comment (pieleric) [inline scripts/spot-grid.py:196]: Please take a DataArray as input. Let's separate reading away.

File: scripts/spot-grid.py [hunk 2]
```diff
@@ -168,6 +191,27 @@ def image_update(imp, window):
...
+        super(StaticImageComponent, self).__init__(name=filename, role='ccd')
```
Comment (pieleric) [inline scripts/spot-grid.py:197]: As name, just hard code to something like: "FakeCCD".

File: scripts/spot-grid.py [hunk 3]
```diff
@@ -224,7 +271,11 @@ def main(args):
     handler.setFormatter(logging.Formatter('%(asctime)s (%(module)s) %(levelname)s: %(message)s'))
     logging.getLogger().addHandler(handler)
 
-    if options.role:
+    if options.filename:
+        logging.info("Will process image file %s" % options.filename)
+        image = StaticImageComponent(options.filename)
```
Comment (pieleric) [inline scripts/spot-grid.py:276]: Let's rename image -> fakeccd.

File: scripts/spot-grid.py [hunk 4]
```diff
@@ -224,7 +271,11 @@ def main(args):
     handler.setFormatter(logging.Formatter('%(asctime)s (%(module)s) %(levelname)s: %(message)s'))
     logging.getLogger().addHandler(handler)
 
-    if options.role:
+    if options.filename:
+        logging.info("Will process image file %s" % options.filename)
```
Comment (pieleric) [inline scripts/spot-grid.py:275]: To open the file, you can do it without skimage, just with what we already have in Odemis. You can write:
``` python
    converter = dataio.find_fittest_converter(fn, default=None, mode=os.O_RDONLY)
    data = converter.read_data(filename)
    image = data[0]
```

File: scripts/spot-grid.py [hunk 5]
```diff
@@ -168,6 +191,27 @@ def image_update(imp, window):
         logging.debug("Display thread ended")
 
 
+class StaticImageComponent(model.DigitalCamera):
```
Comment (pieleric) [inline scripts/spot-grid.py:194]: Can you rename it to StaticCCD?

File: scripts/spot-grid.py [hunk 6]
```diff
@@ -14,6 +14,7 @@
 import cairo
 import logging
 import numpy
+import skimage.io
```
Comment (pieleric) [inline scripts/spot-grid.py:17]: So far we don't depend on skimage. Let's try to not depend on it. If you need more than TIFF + HDF5, you could have a look at Pillow (which we already depend on).

**Review by pieleric — CHANGES_REQUESTED** [review]

**Review by pieleric — APPROVED** [review]

**Review by tepals — APPROVED** [review]

**Pull request merged**